src/fw/flash_region/flash_region_gd25lq255e.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2025 Core Devices LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#define PAGE_SIZE_BYTES (0x100)
+
+#define SECTOR_SIZE_BYTES (0x10000)
+#define SECTOR_ADDR_MASK (~(SECTOR_SIZE_BYTES - 1))
+
+#define SUBSECTOR_SIZE_BYTES (0x1000)
+#define SUBSECTOR_ADDR_MASK (~(SUBSECTOR_SIZE_BYTES - 1))
+
+// A bit of preprocessor magic to help with automatically calculating flash region addresses
+//////////////////////////////////////////////////////////////////////////////
+
+#define FLASH_REGION_DEF(MACRO, arg)                                                      \
+  MACRO(FIRMWARE_SCRATCH, 0x0100000 /*  1024k */, arg)        /*      0x0 - 0x0100000 */  \
+  MACRO(SYSTEM_RESOURCES_BANK_0, 0x0080000 /*   512K */, arg) /* 0x0100000 - 0x0180000 */ \
+  MACRO(SYSTEM_RESOURCES_BANK_1, 0x0080000 /*   512K */, arg) /* 0x0180000 - 0x0200000 */ \
```

Comment:
You'll probably want to make this bigger given your flash is bigger, same for filesystem.

Reply:
Agreed, we'll likely review this in the near future, just expanded file system to cover for the whole flash.